proposals/csharp-9.0/unconstrained-type-parameter-annotations.md

```diff
@@ -0,0 +1,67 @@
+﻿# Unconstrained type parameter annotations
+
+## Summary
+
+Allow nullable annotations for type parameters that are not constrained to value types or reference types: `T?`.
+```C#
+static T? FirstOrDefault<T>(this IEnumerable<T> collection) { ... }
+```
+
+## `?` annotation
+
+In C#8, `?` annotations could only be applied to type parameters that were explicitly constrained to value types or reference types.
+In C#9, `?` annotations can be applied to any type parameter, regardless of constraints.
+
+If a type parameter `T` is substituted with a reference type, then `T?` represents a nullable instance of that reference type.
+If `T` is substituted with a value type, then `T?` is represents an instance of `T`.
```

Comment:
Question: in case of a value type, `T?` represents an instance of `T` or the corresponding nullable `T?` type?